packages/hebao_v1/contracts/modules/base/MetaTxAware.sol

```diff
@@ -30,7 +30,7 @@ abstract contract MetaTxAware
 
     modifier txAwareHashNotAllowed()
     {
-        require(txAwareHash() == 0, "INVALID_TX_AWARE_HASH");
+        //require(txAwareHash() == 0, "INVALID_TX_AWARE_HASH");
```

Comment:
I think this cannot be removed.

Reply:
> I knew this was possible, but I don't see it as a security issue. It's the user that signs the wrong data. We can't really guard against the user signing the wrong things.

> That's why we sign the function selector for the meta tx so it's the user that remains in control and the relayer can't mess things up agains the users. If we check this everywhere (quite error prone because easy to forget) I don't think we even have to sign the function selector like we do now.


If the function that was called actually uses the `txAwareHash` than replay protection is provided `verifyAndUpdate`. 

But if the user signed a meta transaction with `nonce == 0 && txAwareHash != 0` for a function call that doesn't support `txAwareHash` to begin with than you're right that the meta tx can be reused without the check. But why did the user sign an invalid meta tx to begin with, the meta tx would never be valid with the check, not even the first time?

But if you think we need to guard the user from signing an invalid meta tx we can of course keep it, but if the user only signs valid meta txs than I think there is no problem.

Reply:
The extra cost is only 1,200 gas, mainly because of the SLOAD of `trustedForwarder`, which is actually already read by `onlyFromWalletOrOwnerWhenUnlocked` so could be reduced to lower the cost to 400 gas, but it's a bit awkward to cache `trustedForwarder` between the modifiers.

Reply:
Lets keep it please.

Reply:
I reverted the change to txAwareHashNotAllowed